includes/class-wc-install.php

```diff
@@ -565,7 +565,24 @@ private static function create_tables() {
 
 		// Add constraint to download logs if the columns matches.
 		if ( ! empty( $download_permissions_column_type ) && ! empty( $download_log_column_type ) && $download_permissions_column_type === $download_log_column_type ) {
-			$wpdb->query( "ALTER TABLE {$wpdb->prefix}wc_download_log ADD FOREIGN KEY (permission_id) REFERENCES {$wpdb->prefix}woocommerce_downloadable_product_permissions(permission_id) ON DELETE CASCADE" );
+			$wpdb->query( "
+				IF NOT EXISTS (
+					SELECT NULL FROM information_schema.TABLE_CONSTRAINTS
+					WHERE
+						CONSTRAINT_SCHEMA = '{$wpdb->dbname}' AND
+						(
+							CONSTRAINT_NAME LIKE '%wc_download_log_ib%' OR
```

Comment:
Just a minor detail, but at least on my local environment we could be more specific and search for CONSTRAINT_NAME like `{$wpdb->prefix}wc_download_log_ibfk_%` instead of `%wc_download_log_ib%`. Although, I agree that `%wc_download_log_ib%` is probably specific enough so feel free to ignore this comment.

Reply:
I will add a cleanup routine

Reply:
> @rodrigoprimo I did that on purpose for cases where you would restore a DB from another server and the prefix is different.

Ok, makes sense. Thanks for the clarification.

> I will add a cleanup routine

Great, I will check the new version of this PR with the cleanup routine.